common/src/main/java/com/viaversion/viaversion/protocols/protocol1_20_5to1_20_3/rewriter/EntityPacketRewriter1_20_5.java

```diff
@@ -135,15 +138,25 @@ public void registerPackets() {
                     registryEntries[id] = new RegistryEntry(name, tag.get("element"));
                 }
 
-                // Add spit damage type
                 if (Key.stripMinecraftNamespace(type).equals("damage_type")) {
+                    // Add spit damage type
                     final int length = registryEntries.length;
                     registryEntries = Arrays.copyOf(registryEntries, length + 1);
                     final CompoundTag spitData = new CompoundTag();
                     spitData.putString("scaling", "when_caused_by_living_non_player");
                     spitData.putString("message_id", "mob");
                     spitData.putFloat("exhaustion", 0.1F);
                     registryEntries[length] = new RegistryEntry("minecraft:spit", spitData);
+
+                    // Fill in missing damage types with 1.20.3/4 defaults
+                    final Set<String> registryEntryKeys = Arrays.stream(registryEntries).map(RegistryEntry::key).collect(Collectors.toSet());
```

Comment:
In the map call you need to add Key.stripmcnamespace

Reply:
Fixed